exporter/elasticsearchexporter/model.go

```diff
@@ -155,6 +161,131 @@ func (m *encodeModel) encodeLogECSMode(resource pcommon.Resource, record plog.Lo
 	return document
 }
 
+func (m *encodeModel) encodeMetrics(resource pcommon.Resource, metrics pmetric.MetricSlice, _ pcommon.InstrumentationScope) ([][]byte, error) {
+	hasher := sha256.New()
+	var baseDoc objmodel.Document
+
+	baseDoc.AddAttributes("", resource.Attributes())
+
+	// Put all metrics that have the same attributes and timestamp in one document.
+	docs := map[string]*objmodel.Document{}
+	for i := 0; i < metrics.Len(); i++ {
+		metric := metrics.At(i)
+
+		var dps pmetric.NumberDataPointSlice
+
+		// Only Gauge and Sum metric types are supported at the moment.
+		switch metric.Type() {
+		case pmetric.MetricTypeGauge:
+			dps = metric.Gauge().DataPoints()
+		case pmetric.MetricTypeSum:
+			dps = metric.Sum().DataPoints()
+		}
+
+		for j := 0; j < dps.Len(); j++ {
+			dp := dps.At(j)
+
+			hash := metricHash(hasher, dp.Timestamp(), dp.Attributes())
+			doc, docExists := docs[hash]
+			if !docExists {
+				doc = baseDoc.Clone()
+				doc.AddTimestamp("@timestamp", dp.Timestamp())
+				doc.AddAttributes("", dp.Attributes())
+
+				docs[hash] = doc
+			}
+
+			switch dp.ValueType() {
+			case pmetric.NumberDataPointValueTypeDouble:
+				doc.AddAttribute(metric.Name(), pcommon.NewValueDouble(dp.DoubleValue()))
+			case pmetric.NumberDataPointValueTypeInt:
+				doc.AddAttribute(metric.Name(), pcommon.NewValueInt(dp.IntValue()))
+			}
+		}
+	}
+
+	res := make([][]byte, 0, len(docs))
+
+	for _, doc := range docs {
+		if m.dedup {
+			doc.Dedup()
+		} else if m.dedot {
+			doc.Sort()
+		}
+
+		var buf bytes.Buffer
+		err := doc.Serialize(&buf, m.dedot)
+
+		if err != nil {
+			fmt.Printf("Serialize error, dropping doc: %v\n", err)
```

Comment:
Should we be returning the error?

Reply:
I changed this code to return the error: https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/33513/commits/2e9cfb82aef46db9d5bbaf6c514e0cca39b2fc00.